src/ir/effects.h

```diff
@@ -444,7 +423,7 @@ struct EffectAnalyzer
     }
   }
   void visitBinary(Binary* curr) {
-    if (!ignoreImplicitTraps) {
+    if (!ignoreImplicitTraps && !implicitTrap) {
```

Comment:
How about removing this line entirely instead? (and making the necessary changes below)

Reply:
At least we need `if (implicitTrap)` otherwise we should do union like `implicitTrap |= ...`

Reply:
Or do `if (condition) implicitTrap = true;` later. I think this would be a nice simplifying change in line with many of the other changes in this PR.

Reply:
I still think early check better due to if `implicitTrap` already set we don't need do whole switch and other checks

Reply:
This is something that the compiler could do if it's worthwhile - it can see that the switch just leads to a single assignment. So it can optimize this for size or speed accordingly. I think this addition won't help, and makes the code a little less readable.